### PR TITLE
Update to redis v0.24

### PR DESCRIPTION
**Pull request by aumetra**
Updates the version of `redis` used by `deadpool-redis` to v0.24

### CODE REVIEW

**Comment by bikeshedder** [comment]
LGTM. I just want to make a `deadpool-redis-cluster` release with an updated README stating that this crate is deprecated.

I do wonder if I should release a `deadpool-redis-cluster 0.1.1` version that just states the deprecation or a version `0.2` which doesn't contain any code at all. The former is just a tiny nudge while the latter is a bit "aggresive" and will let anyone using the crate know about that when trying to upgrade to version `0.2`.

**Comment by brocaar** [comment]
@bikeshedder, this comment seems to be related to #287, not the update to redis 0.24 PR... I'll comment there...

**Comment by bikeshedder** [comment]
> @bikeshedder, this comment seems to be related to #287, not the update to redis 0.24 PR... I'll comment there...

Whops. Yes. :facepalm:

File: redis/src/config.rs [hunk 1]
```diff
@@ -124,7 +124,8 @@ impl Default for Config {
     }
 }
 
-/// This is a 1:1 copy of the [`redis::ConnectionAddr`] enumeration.
+/// This is a 1:1 copy of the [`redis::ConnectionAddr`] enumeration (excluding `tls_params` since it is entirely opaque to consumers).
```
Comment (bikeshedder) [inline redis/src/config.rs:127]: Are you sure about that? Maybe the `tls_params` should be exposed by our config struct as well? I feel a bit uneasy leaving out configuration fields. Even if those fields can't be deserialized with serde it should be possible to set the `tls_params` after deserialization in code.

Comment (aumetra) [reply redis/src/config.rs:127]: Why I believe that it's an oversight is that those fields here are exposed: <https://github.com/redis-rs/redis-rs/blob/3e6dc2ef611060a665d8e5d4c14bd8618517fac2/redis/src/lib.rs#L468-L469>

But yet the actual `TlsConnParams` is kept opaque.

Comment (aumetra) [reply redis/src/config.rs:127]: I created an issue on their issue tracker to hopefully clarify this: https://github.com/redis-rs/redis-rs/issues/1006

**Pull request merged**